### PR TITLE
deprecate pickleable sessions, recommend json

### PR DESCRIPTION
**Pull request by mmerickel**
This change is backward-compatible but emits deprecation warnings when using the default `PickleSerializer` with `SignedCookieSessionFactory`.

I was also going to deprecate it in `UnencryptedCookieSessionFactoryConfig` but that entire thing has been deprecated since 1.5 and is definitely getting removed in 2.0 wholesale, so there's no point in deprecating a small part of it.

replaces #3341 
related #2709

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -57,6 +57,11 @@ Features
 - Add support for Python 3.7. Add testing on Python 3.8 with allowed failures.
   See https://github.com/Pylons/pyramid/pull/3333
 
+- Added ``pyramid.session.JSONSerializer``. See "Upcoming Changes to ISession
+  in Pyramid 2.0" in the "Sessions" chapter of the documentation for more
+  information about this feature.
```
[the remark this answers is not in the document]
Comment (mmerickel) [reply CHANGES.rst]: The CHANGES.rst is not part of the docs and as such should not try to do any internal links. That's what the whatsnew doc is for. CHANGES.rst primarily shows up on the pypi page. We could make an absolute link but I'd rather not do it. There will be a good link in the actual whatsnew document when we release things.

**Comment by mmerickel** [comment]
> i think it would make sense for the docs to list some of the common objects that not JSON serializable (namely Datetime) and show an example JSON encoder/decoder using object_hooks to enable it.

I'm not interested in doing this. Feel free to submit a PR and I'll evaluate it but I have no intentions of supporting this type of stuff in the default json serializer on my own.

**Comment by stevepiercy** [comment]
@jvanasco your PR may be useful as a recipe in the Pyramid Community Cookbook, if not in Pyramid's docs, or you could blog about it, and Pyramid docs could have a link to it.

**Comment by jvanasco** [comment]
ok, i'll generate a PR for some docs once this ticket closes out.  I don't think the default serializer should handle `datetime`/etc natively; but i do think the docs should instruct users on how they can extend the session serializer to handle other objects - especially if `pickle` is being presented as a fallback.

**Pull request merged**